Sources/MapboxCoreNavigation/Router.swift

```diff
@@ -39,6 +39,49 @@ public struct IndexedRouteResponse {
         return routeResponse.routes?[routeIndex]
     }
     
+    /**
+     Parses routes in the current response, accounting `routeIndex` to be pointing to the primary route, while all the others being `AlternativeRoute`s.
+     
+     - returns: Array of `AlternativeRoutes` containing relative information to `currentRoute`.
+     */
+    public func parseAlternativeRoutes() -> [AlternativeRoute] {
+        guard let mainRoute = currentRoute,
+              let routesData = routesData(routeParserType: RouteParser.self) else {
+            return []
+        }
+        
+        let alternatives = routesData.alternativeRoutes().compactMap {
+            AlternativeRoute(mainRoute: mainRoute,
+                             alternativeRoute: $0)
+        }
+        return alternatives
+    }
+    
+    func routesData(routeParserType: RouteParser.Type) -> RoutesData? {
```

Comment:
Would it be possible to add new tests for this non-private function? To test the case when `routeParser` returned error

Reply:
I am not sure it is needed, as it is 90% a wrapper method